ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala

```diff
@@ -225,8 +225,12 @@ object KeyValueCommitting {
       node.getNodeTypeCase match {
 
         case TransactionOuterClass.Node.NodeTypeCase.ROLLBACK =>
-          // TODO https://github.com/digital-asset/daml/issues/8020
-          sys.error("rollback nodes are not supported")
+        // Rollback nodes themselves do not produce any outputs. However, children
+        // of rollback nodes potentially will, e.g., divulgence.
+        // At the moment, we overapproximate and treat
+        // a node the same regardless of whether it was under a rollback node.
+        // This matches the treatment of transient contracts which could also
+        // be trimmed from the output.
```

Comment:
Something like the following gives more context IMO:

```suggestion
        // Nodes under rollback will potentially produce outputs such as divulgence.
        // Actual outputs must be a subset of, or the same as, computed outputs and
        // we currently relax this check by widening the latter set, treating a node the
        // same regardless of whether it was under a rollback node or not.
        // Computed outputs that are not actual outputs can be safely trimmed: examples
        // are transient contracts and, now, potentially also outputs from nodes under
        // rollback.
```

Reply:
This might have a performance impact BTW.

Reply:
Definitely but the same is true for transient contracts. I don’t think it makes much sense to treat transient contracts differently from rollback node. If the performance matters here, we should filter out both or keep it simple and filter out neither.

Reply:
Agreed, it is more of  a note to self so that (perhaps) we'll know what to do (i.e., trim more) if we see a performance degradation with upgrades to the next SDK.

Reply:
You should only see a degredation once someone uses exceptions and at that point it’s probably hard to diff since other things changed. I think if you want to see the performance impact you need to test the same code that uses exceptions and run it once with trimmed outputs and once without.